src/rules/custom-property-empty-line-before/README.md

```diff
@@ -0,0 +1,154 @@
+# custom-property-empty-lines-before
+
+Require or disallow an empty line before custom properties.
+
+```css
+a {
+  --custom-prop: value;
+                          /* ← */
+  --custom-prop2: value;  /* ↑ */   
+}                         /* ↑ */
+/**                          ↑
+ *                   This line */
+```
+
+## Options
+
+### `always`
+
+The following patterns are considered warnings:
+
+```css
+a {
+  --custom-prop: value;
+  --custom-prop2: value;
+}
+```
+
+```css
+a { --custom-prop: value;
+  --custom-prop2: value;
+}
+```
+
+The following patterns are *not* considered warnings:
+
+```css
+a {
+
+  --custom-prop: value;
+
+  --custom-prop2: value;
+}
+```
+
+### `never`
+
+The following patterns are considered warnings:
+
+```css
+a {
+  --custom-prop: value;
+
+  --custom-prop2: value;
+}
+```
+
+```css
+a {
+
+  --custom-prop: value;
+  --custom-prop2: value;
+}
+```
+
+The following patterns are *not* considered warnings:
+
+```css
+a {
+  --custom-prop: value;
+  --custom-prop2: value;
+}
+```
+## Optional options
+
+### `expect: ["first-nested", "after-comment", "after-custom-property"]`
+
+### `first-nested`
+
+Reverse the primary option for custom-property that are nested.
```

Comment:
"Reverse the primary option for custom-properties that are are nested and the first child of their parent node."

We say this [else where](https://github.com/stylelint/stylelint/blob/master/src/rules/at-rule-empty-line-before/README.md#first-nested)